### DISCUSSION TITLE
Disable docker pinDigests when currentValue is "latest"

### DISCUSSION THREAD

**Discussion by dnwe** [renovatebot/renovate#22720] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.98.1

### If you're self-hosting Renovate, select which platform you are using.

GitHub Enterprise Server

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Similar to https://github.com/renovatebot/renovate/discussions/20614 I'd like to selectively disable docker pinDigests — but in my case I wanted to do it based on matching the currentValue of "latest"

### What you tried so far.

I tried at the top-level in config.js:
```js
    docker: {
      extends: ["docker:enableMajor", "docker:pinDigests"],
      packageRules: [
        {
          matchCurrentValue: "latest", // don't pin digests on the 'latest' tag
          pinDigests: false,
        }
      ]
    },
```

But still see PR's raised to pin the image tag by appending `@sha256:xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx`

Before I dig in to the debug logs, there's nothing implicitly wrong with this approach is there?

**Comment by viceice** [comment]
you can't nest package rules inside of other objects, it's a global config option. so move it to the root level of the config.

**Reply by viceice** [reply]
same for your extends

**Reply by dnwe** [reply]
Interesting. `renovate-config-validator` doesn't pick that up and seemed to parse and be happy with the nesting (once I changed `latest` to `/latest/`)

```
$ renovate-config-validator
 INFO: Validating config.js
ERROR: config.js contains errors
       "errors": [
         {
           "topic": "Configuration Error",
           "message": "Invalid regExp for docker.packageRules[0].matchCurrentValue: `latest`"
         }
       ]
$ sed -e 's|latest|/latest/' -i config.js
$ renovate-config-validator
 INFO: Validating config.js
 INFO: Config validated successfully

```

**Reply by dnwe** [reply]
I also note that there are other nested packageRules in my config which do seem to currently work and get applied correctly
e.g.,
```js
"golang": {
  "packageRules": [
    {
      "description": [
        "Disable major/minor/digest updates for k8s.io packages"
      ],
      "matchPackageNames": [
        "k8s.io/api",
        "k8s.io/apiextensions-apiserver",
        "k8s.io/apimachinery",
        "k8s.io/apiserver",
        "k8s.io/cli-runtime",
        "k8s.io/client-go",
        "k8s.io/kubectl"
      ],
      "matchUpdateTypes": ["major", "minor", "digest"],
      "enabled": false
    },
    {
      "description": ["Permit and group patch updates for k8s.io packages"],
      "matchPackageNames": [
        "k8s.io/api",
        "k8s.io/apiextensions-apiserver",
        "k8s.io/apimachinery",
        "k8s.io/apiserver",
        "k8s.io/cli-runtime",
        "k8s.io/client-go",
        "k8s.io/kubectl"
      ],
      "matchUpdateTypes": ["patch"],
      "groupName": "k8s.io modules"
    }
  ]
}
```

**Reply by rarkins** [reply]
It can work, but we strongly recommend against it, especially because the validator can't always pick up which types work and which don't
